src/v/storage/disk_log_impl.cc

```diff
@@ -871,7 +871,7 @@ disk_log_impl::make_reader(timequery_config config) {
             0,
             2048, // We just need one record batch
             cfg.prio,
-            std::nullopt,
+            {model::record_batch_type::raft_data},
```

Comment:
timequery is indeed used only by Kafka layer, but maybe for completeness we can add `batch_type` filter to `timequery_config` ?